gnosis/safe-react
# #80 Send funds for threshold = 1

**THIS PR**
- Implements Send Funds modal
- Adds logic for calculating gas (`logic/safe/transactions/gas.js`). This is not used anywhere yet, but just in case.
- Adds more functionality to layout components (mutators to form, passing classes to select fields, setting minHeight for button, passing styles to Hairline, etc)
- Added `why-did-you-render` dev dependency to debug avoidable re-renders:
https://github.com/welldone-software/why-did-you-render
- Adds `ButtonLink` layout component - a Button styled as link
- Fixes some mui styles
- Dependencies bump
- Minor codestyle fixes
- Migrate to Material UI v4

**TESTING:**
- Go to your safe
- Select an asset in your balances table, click send on it, a modal should pop up with this asset pre-selected
- Try to close modal and open it again, it should open with no values prepopulated (except Token)
- Try to input invalid data (balance greater than yours, non ethereum address)
- With valid data, click review
- Try to click back, you should see the previous form, all the data should be saved
- Click review again
- Click send
- Modal should close, you should see a metamask pop-up, click accept
- After some time, you should see a notification: 'Tx has been successfully submitted'

**THINGS TO DICUSS:**
What should be the flow after clicking 'Send' in final Review screen?
Currently it is:
1. "Review" button is clicked
2. Modal is closed
3. Metamask pop-up opens
4. After tx is confirmed, a notification pops up

Should it be 1, 3, 2, 4? Should we maybe split notifications to 'Tx has been successfully submitted' and 'Tx has been successfully confirmed'?
